src/compiler/utilitiesPublic.ts

```diff
@@ -761,6 +762,15 @@ export function unescapeLeadingUnderscores(identifier: __String): string {
 export function idText(identifierOrPrivateName: Identifier | PrivateIdentifier): string {
     return unescapeLeadingUnderscores(identifierOrPrivateName.escapedText);
 }
+
+/**
+ * If the text of an Identifier matches a keyword (including contextual and TypeScript-specific keywords), returns the
+ * SyntaxKind for the matching keyword.
+ */
+export function idKeyword(node: Identifier) {

```

Comment:
I would prefer we call this `identifierToKeywordKind` or something like that.

Reply:
Sure. I just named it that way to match `idText` so that they would be sorted together in completions. The FP-like design of the compiler generally lends itself to using common prefixes to group related functionality together.

Reply:
I may just use a more general name like `identifierToToken`, since we don't necessarily validate identifier text, nor do we check that the result is limited to only keyword tokens.